.github/workflows/ci.yaml

```diff
@@ -97,13 +119,45 @@ jobs:
           path: |
             ~/.cargo/
             target/
+            .venv/
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Python Bindings Tests
         run: |
-          just python build
           just python test
 
+  benchmarks:
+    name: Benchmarks (Scale Factor = ${{ matrix.scale_factor }})
+    runs-on: ubuntu-latest-8-cores
+    needs: ["build-python"]
+    # CI on push is only triggered for main branch.
+    if: github.event_name == 'push'
+    strategy:
+      # We want all the benchmark jobs to run one by one.
+      max-parallel: 1
+      matrix:
+        scale_factor: [1]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Required to generate TPC-H data
+          submodules: 'recursive'
+      - uses: extractions/setup-just@v1
+      - uses: actions/cache@v3
+        name: Cache
+        with:
+          path: |
+            ~/.cargo/
+            target/
+            .venv/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Run benchmarks
+        run: |
+          just bench tpch gen_scale_factor ${{ matrix.scale_factor }}
+          # Run the benchmarks 3 times.
+          just bench tpch run 3 ${{ matrix.scale_factor }}
```

Comment:
why 3?t

Reply:
TBH, I had this in the matrix but later realized we'd probably end up fixing a number. I don't have a genuine reason for `3`, but it seemed like a decent number for getting the average timing (when we analyze the data). We may want to change this based on the scale factor. Any thoughts? I'm thinking of more runs for a smaller scale factor. But this is just because we can, I don't think that would be required.

Reply:
Oh, I can see arguments for 4 or 5 too, just wanted to know why we picked this :)